update: thelounge configuration example

ishanjain28 opened the pull request.

nitnelave commented on example_configs/thelounge.md:
```diff
@@ -5,12 +5,19 @@
+        tlsOptions: {},
+        primaryKey: "uid",
+        searchDN: {
+            rootDN: "uid=ldap-editor,ou=people,DC=example,DC=com",
```
What's the root DN? And why ldap-editor? Maybe we need to describe this a bit more

ishanjain28 replied:
rootDN/rootPassword are the credentials used to login to lldap and perform queries. (Similar to bind user and bind user password. the lounge uses a different name for it) 

`ldap-editor` user is a member of `lldap_strict_readonly` and `lldap_password_manager` groups.

nitnelave replied:
Right, this would be great to put in the documentation :)

ishanjain28 commented:
hey @nitnelave Please let me know if you have any more feedback on this PR. It is complete from my end and awaiting your approval

nitnelave commented:
@ishanjain28  Sorry, my comment was stuck in "pending", I never actually sent it.

ishanjain28 commented on example_configs/thelounge.md:
````diff
@@ -5,12 +5,22 @@
+        },
+    },
 ```
 
````
That makes sense and I'll make this change.

nitnelave commented:
Hey @ishanjain28 , do you want to update this PR, or should I take it over?

nitnelave commented on example_configs/thelounge.md:
```diff
@@ -2,15 +2,39 @@
 [Main documentation here.](https://thelounge.chat/docs/configuration#ldap-support)
 
-For simple user auth with LLDAP on localhost adapt this in the main config.js:
+Simple Config:
```
Make it a title, with 2 `#` before

nitnelave commented on example_configs/thelounge.md:
````diff
@@ -2,15 +2,39 @@
 ```
 
-And comment out with ```//``` the entire ```searchDN``` block.
+In this config, The Lounge will use the credentials provided in web ui to authenticate with lldap. It'll allow access if authentication was successful and the user is a member of Base DN.
````
I would put the description first, before the config.

Also, all users are members of the base DN, so we don't need to specify that.

nitnelave commented on example_configs/thelounge.md:
```diff
@@ -2,15 +2,39 @@
+In this config, The Lounge will use the credentials provided in web ui to authenticate with lldap. It'll allow access if authentication was successful and the user is a member of Base DN.
+
+
+Advanced Config:
```
Title

nitnelave commented on example_configs/thelounge.md:
````diff
@@ -2,15 +2,41 @@
+    },
+```
+
+`rootDN` is similar to bind DN in other applications. It is used in combination with `rootPassword` to query lldap. `ldap-viewer` user in `lldap` is a member of the group `lldap_strict_readonly` group. This gives `ldap-viewer` user permission to query `lldap`.
````
Can you reset your password via the lounge? Also, the readonly permission is redundant, password manager is a superset.

ishanjain28 replied:
@nitnelave Yep, Updated this in the next commit

ishanjain28 commented:
Hey @nitnelave Apologies for the delays. 

I have updated the config into configs for simple and more advanced setups and rebased the PR. 

> The root DN is the bind user, preferably a member of strict readonly.

This was a great suggestion! I assumed the lounge supported changing passwords for users but they do not so we should not use a user with the permission to change passwords in this context, Just the user with read permissions is sufficient.

ishanjain28 commented:
I believe all of the comments have been addressed. I am not sure why github's UI is still stuck with "1 change requested"

nitnelave reviewed (approved):
Great, thanks!

nitnelave commented:
It was just waiting for my approval :)

nitnelave commented:
Thank you everyone for the contribution!
This feature is now available in the latest release, [v0.5.0](https://github.com/lldap/lldap/releases/tag/v0.5.0).
You can support LLDAP by starring our repo, contributing some configuration examples and becoming a sponsor.
